### PR TITLE
dkb adapter

### PR DESCRIPTION
**Pull request by spetrychenko**

### CODE REVIEW

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            accessToken = new AccessToken(response.getResponseBody());
+        }
+        return accessToken.token;
+    }
+
+    void setHttpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    private boolean isValid() {
```
Comment (n0rdy) [inline adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: What do you think about renaming this method to `isNotValid()` to increase the code readability? As far as I see, now it is used for negative comparison only (`!isValid()`).

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                DEFAULT_SECONDS_BEFORE_TOKEN_EXPIRATION
+        ));
+    }
+
+    @Override
+    public String retrieveToken() {
+        if (!isValid()) {
+            GeneralResponse<TokenResponse> response = httpClient.post(
+                    tokenUrl,
+                    "grant_type=client_credentials",
```
Comment (n0rdy) [inline adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: Could you, please, extract this String to some constants to increase the code readability?

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+
+    private DkbAccessTokenService() {
+        jsonMapper = new JsonMapper();
+        setHttpClient(HttpClient.newHttpClient());
+    }
+
+    public static AccessTokenService getInstance() {
+        if (instance == null) {
+            instance = new DkbAccessTokenService();
```
Comment (n0rdy) [inline adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: What do you think: shouldn't we be afraid of race conditions here? I mean, it is possible, that more than 1 instances could be created. I think, maybe lazy initialization is not needed here. What is your opinion?

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    static class TokenResponse {
+        // {"access_token":"9c60cc0a-7b38-3a2a-9b43-2b7d9d957838","scope":"am_application_scope default","token_type":"Bearer","expires_in":3600}
+        @JsonProperty("access_token")
+        private String accessToken;
+        @JsonProperty("expires_in")
+        private int expiresIn;
+
+        TokenResponse() {
```
Comment (n0rdy) [inline adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: Could you, please, leave a comment for the future generations, that this constructor is needed for the Jackson lib? IntelliJ proposes to remove it, so I propose to leave a few words, why not. Thanks!

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        }
+
+        TokenResponse(String accessToken, int expiresIn) {
+            this.accessToken = accessToken;
+            this.expiresIn = expiresIn;
+        }
+    }
+
+    private static class AccessToken {
+        private String token = "";
```
Comment (n0rdy) [inline adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: The initialization of this variable is redundant.

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java (hunk 4 above)
Comment (spetrychenko) [reply adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: Thanks, have added a comment

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java (hunk 5 above)
Comment (spetrychenko) [reply adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: Thanks, removed

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java (hunk 1 above)
Comment (spetrychenko) [reply adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: Thanks, could be better for code readability

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java (hunk 2 above)
Comment (spetrychenko) [reply adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: Thanks, have moved to constants

File: adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java (hunk 3 above)
Comment (spetrychenko) [reply adapters/dkb-adapter/src/main/java/de/adorsys/xs2a/gateway/service/impl/DkbAccessTokenService.java]: Thanks, it makes sense

**Pull request merged**